src/conn/conn_open.c

```diff
@@ -198,33 +198,42 @@ __wt_connection_close(WT_CONNECTION_IMPL *conn)
 int
 __wt_connection_workers(WT_SESSION_IMPL *session, const char *cfg[])
 {
+    WT_DECL_RET;
+
     /*
      * Start the optional statistics thread. Start statistics first so that other optional threads
      * can know if statistics are enabled or not.
      */
     WT_RET(__wt_statlog_create(session, cfg));
     WT_RET(__wt_logmgr_create(session, cfg));
 
+    /* Initialize metadata tracking, required before creating tables. */
+    WT_RET(__wt_meta_track_init(session));
+
+    /* Try to create the history store table. */
+    ret = __wt_hs_create(session, cfg);
+
+    if (ret != 0 && ret != ENOENT && ret != WT_NOTFOUND)
```

Comment:
Can we add a comment to explain the reason behind those error cases of why they are expected and in what scenarios?

Reply:
I actually don't know. I just follow what is instructed in the ticket. Can you add the comment @luke-pearson ?

Reply:
Added comment.